### ISSUE TITLE
ADIOS 2.5.0 Warning: BPSerialiser Instantiation-after-Specialization

### ISSUE THREAD

**Issue by ax3l** [ornladios/ADIOS2#1808]
With ADIOS 2.5.0 warning with `clang@5.0.0` and `-Winstantiation-after-specialization`:

```
source/adios2/toolkit/format/bp/BPSerializer.cpp:875:39:
error: explicit instantiation of
'PutAttributeCharacteristicValueInIndex<std::__cxx11::basic_string<char> >'
that occurs after an explicit specialization has no effect [-Winstantiation-after-specialization]
ADIOS2_FOREACH_ATTRIBUTE_STDTYPE_1ARG(declare_template_instantiation)

source/adios2/toolkit/format/bp/BPSerializer.inl:23:27:
note: previous template specialization is here
inline void BPSerializer::PutAttributeCharacteristicValueInIndex(

source/adios2/toolkit/format/bp/BPSerializer.cpp:875:39:
error: explicit instantiation of
'GetAttributeSizeInData<std::__cxx11::basic_string<char> >'
that occurs after an explicit specialization has no effect [-Winstantiation-after-specialization]

ADIOS2_FOREACH_ATTRIBUTE_STDTYPE_1ARG(declare_template_instantiation)
source/adios2/toolkit/format/bp/BPSerializer.inl:68:29:
note: previous template specialization is here
inline size_t BPSerializer::GetAttributeSizeInData(source/adios2/toolkit/format/bp/BPSerializer.cpp:898:29: error: explicit instantiation of
'PutPayloadInBuffer<std::__cxx11::basic_string<char> >'
that occurs after an explicit specialization has no effect
[-Winstantiation-after-specialization]

ADIOS2_FOREACH_STDTYPE_1ARG(declare_template_instantiation)
source/adios2/toolkit/format/bp/BPSerializer.inl:58:27:
note: previous template specialization is here
inline void BPSerializer::PutPayloadInBuffer(
```

**Comment by williamfgc** [comment]
@ax3l thanks, I might have to do extra cleaning after the refactoring.

**Comment by williamfgc** [comment]
@ax3l this should be fixed on master. Thanks!

**Comment by ax3l** [comment]
Thanks a lot, fixed with #1865.
